device/device.go

```diff
@@ -177,9 +177,35 @@ func (d *Device) IsConnected() bool {
 	return false
 }
 
-// AddInterface adds an interface to the device
-func (d *Device) AddInterface(astarteInterface interfaces.AstarteInterface) {
+// AddInterface adds an interface to the device. It assumes that the interface is loaded
+// with ParseInterface from the astarte-go/interfaces package.
+// AddInterface returns `nil` if the interface to be loaded is valid and all its fields
```

Comment:
AddInterface returns `nil` if the interface was loaded successfully, or a corresponding error otherwise (e.g.: the interface failed validation).